appstudio-controller/controllers/appstudio.redhat.com/deploymenttargetbinder_controller.go

```diff
@@ -0,0 +1,374 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appstudioredhatcom
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	applicationv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// DeploymentTargetClaimReconciler reconciles a DeploymentTargetClaim object
+type DeploymentTargetClaimReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/finalizers,verbs=update
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets/status,verbs=get;update;patch
+
+const (
+	annBindCompleted string = "dt.appstudio.redhat.com/bind-complete"
+
+	annBoundByController string = "dt.appstudio.redhat.com/bound-by-controller"
+
+	annTargetProvisioner string = "provionser.appstudio.redhat.com/dt-provisioner"
```

Comment:
nit: typo: `provionser.appstudio.redhat.com/dt-provisioner` -> `provisioner.appstudio.redhat.com/dt-provisioner`

Reply:
Yes, this typo is made a few times in the actual ADR-8 doc :smile:.